toolz/itertoolz.py

```diff
@@ -495,6 +496,21 @@ def concatv(*seqs):
     return concat(seqs)
 
 
+def cycles(n, seq):
+    """ Cycles through the sequence n-times.
+
+    Note: If ``n`` is ``None``, cycle infinitely.
+
+    >>> list(cycles(2, [1, 2, 3]))
+    [1, 2, 3, 1, 2, 3]
+    """
+
+    if n is None:
```

Comment:
I suppose, but I thought it was nice to have this be a generalization on `itertools.cycle` especially since their names are so similar. Plus it is nice not to have to litter other code with similar branches if possible.